### ISSUE TITLE
Implementing `conflicts_with`

### ISSUE THREAD

**Issue by vitorgalvao** [caskroom/homebrew-cask#12822]
Already in casks, but not yet functional. The [previous attempt and ensuing discussion should be taken into consideration](https://github.com/caskroom/homebrew-cask/pull/9319).

**Comment by commitay** [comment]
`conflicts_with :cask` https://github.com/Homebrew/brew/pull/3001

**Comment by commitay** [comment]
Updated the issue to reflect the changes in https://github.com/Homebrew/homebrew-cask/pull/52349

**Comment by FranklinYu** [comment]
Before we implement this, is it helpful to add caveats to casks as poor-man's `conflict_with`?

**Comment by vitorgalvao** [comment]
> Before we implement this, is it helpful to add caveats to casks as poor-man's `conflict_with`?

Not really, no. It’s been without it for years and there has been no issue.

> Will draft a PR.

Thank you. Please report back here when you do.

> I guess it would be useful if `conflicts_with formula` prints something in `brew cask info`?

I have no preference. For keeping consistency with non-Cask Homebrew, there should some info, but `conflicts_with cask:` doesn’t show right now. Either both should show or neither should.

**Comment by osalbahr** [comment]
Is this still needed?

**Comment by bevanjkay** [comment]
@osalbahr Yes it is

**Comment by atyndall** [comment]
The wireshark conflict issue just happened to me, implementing this would be a good idea

**Comment by gouravrc** [comment]
This solved for me [#12822](https://github.com/Homebrew/homebrew-cask/issues/146078#issuecomment-1527357453)

**Comment by nikitabobko** [comment]
`conflicts_with formula:` is being deprecated https://github.com/Homebrew/brew/pull/20499
